src/main/java/org/opensearch/flowframework/workflow/DeployModelStep.java

```diff
@@ -57,8 +55,8 @@ public void onResponse(MLDeployModelResponse mlDeployModelResponse) {
 
             @Override
             public void onFailure(Exception e) {
-                logger.error("Model deployment failed");
-                deployModelFuture.completeExceptionally(e);
+                logger.error("Failed to deploy model");
+                deployModelFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));
```

Comment:
Let's use `ExceptionsHelper.status` here too to set the RestStatus code based on the exception type